src/common/offline_transformations/src/compress_quantize_weigths.cpp

```diff
@@ -211,86 +224,630 @@ ov::pass::CompressQuantizeWeights::CompressQuantizeWeights() {
+        if (!evaluate_node(fq_input, ov::TensorVector{tensor_from_constant(weights)}, weights_tensor))
+            return {};
+    } else {
+        weights_tensor = tensor_from_constant(fq_input_constant);
+    }
+
+    auto in_low = ov::as_type_ptr<ov::op::v0::Constant>(fq->get_input_node_shared_ptr(1));
+    auto in_high = ov::as_type_ptr<ov::op::v0::Constant>(fq->get_input_node_shared_ptr(2));
+    auto out_low = ov::as_type_ptr<ov::op::v0::Constant>(fq->get_input_node_shared_ptr(3));
+    auto out_high = ov::as_type_ptr<ov::op::v0::Constant>(fq->get_input_node_shared_ptr(4));
+
+    return ov::TensorVector{weights_tensor,
+                            tensor_from_constant(in_low),
+                            tensor_from_constant(in_high),
+                            tensor_from_constant(out_low),
+                            tensor_from_constant(out_high)};
+}
+
+template <typename T>
+static std::shared_ptr<ov::Node> get_single_consumer_of_type(const std::shared_ptr<ov::Node>& node) {
+    auto target_inputs = node->output(0).get_target_inputs();
+    if (target_inputs.size() != 1)
+        return nullptr;
+    auto consumer = ov::as_type<T>(target_inputs.begin()->get_node());
+    if (!consumer)
+        return nullptr;
+    return consumer->shared_from_this();
+}
+
+bool has_dequantization_subgraph(const std::shared_ptr<ov::Node>& fq,
+                                 std::shared_ptr<ov::Node>& convert_to_low_precision,
+                                 std::shared_ptr<ov::Node>& convert_to_high_precision,
+                                 std::shared_ptr<ov::Node>& zero_point) {
+    convert_to_low_precision = get_single_consumer_of_type<ov::op::v0::Convert>(fq);
+    if (!convert_to_low_precision)
+        return false;
+    convert_to_high_precision = get_single_consumer_of_type<ov::op::v0::Convert>(convert_to_low_precision);
+    if (!convert_to_high_precision)
+        return false;
+    auto subtract = get_single_consumer_of_type<ov::op::v1::Subtract>(convert_to_high_precision);
+    ov::Node* multiply;
+    if (subtract) {
+        zero_point = subtract->get_input_node_shared_ptr(1);
+        auto subtract_consumers = subtract->output(0).get_target_inputs();
+        if (subtract_consumers.size() != 1)
             return false;
-        auto adj_zero_point_val = adj_zero_point_const->cast_vector<float>();
-        bool is_adj_zero_point_close_to_zero =
-            std::all_of(adj_zero_point_val.begin(), adj_zero_point_val.end(), [](float f) -> bool {
-                return std::fabs(f) < 1e-4;
-            });
-        if (!is_adj_zero_point_close_to_zero)
+        multiply = subtract_consumers.begin()->get_node();
+    } else {
+        auto convert_to_high_precision_consumers = convert_to_high_precision->output(0).get_target_inputs();
+        if (convert_to_high_precision_consumers.size() != 1)
             return false;
+        multiply = convert_to_high_precision_consumers.begin()->get_node();
```

Comment:
Minor: why can't we use `get_single_consumer_of_type` to get multiply?

Reply:
changed the code to use get_single_consumer_of_type